msal/tests/Test.MSAL.NET.Unit.net45/CoreTests/InstanceTests/B2cAuthorityTests.cs

```diff
@@ -79,28 +82,80 @@ public void NotEnoughPathSegmentsTest()
 
         [TestMethod]
         [TestCategory("B2CAuthorityTests")]
-        public void ValidationEnabledNotSupportedTest()
+        public void B2CLoginAuthorityCreateAuthority()
         {
             using (var httpManager = new MockHttpManager())
             {
-                var instance = Authority.CreateAuthority(CoreTestConstants.B2CAuthority, true);
+                //add mock response for tenant endpoint discovery
+                httpManager.AddMockHandler(
+                    new MockHttpMessageHandler
+                    {
+                        Method = HttpMethod.Get,
+                        Url = "https://mytenant.com.b2clogin.com/tfp/mytenant.com/policy/v2.0/.well-known/openid-configuration",
+                        ResponseMessage = MockHelpers.CreateSuccessResponseMessage(
+                           File.ReadAllText(ResourceHelper.GetTestResourceRelativePath("OpenidConfiguration-B2CLogin.json")))
+                    });
+
+                Authority instance = Authority.CreateAuthority("https://mytenant.com.b2clogin.com/tfp/mytenant.com/policy/", true);
                 Assert.IsNotNull(instance);
                 Assert.AreEqual(instance.AuthorityType, AuthorityType.B2C);
-                try
-                {
-                    instance.UpdateCanonicalAuthorityAsync(
-                                    httpManager,
-                                    new TelemetryManager(),
-                                    new RequestContext(null, new TestLogger(Guid.NewGuid(), null))
-                                    ).GetAwaiter().GetResult();
-
-                    Assert.Fail("test should have failed");
-                }
-                catch (ArgumentException exc)
-                {
-                    Assert.IsInstanceOfType(exc, typeof(ArgumentException));
-                    Assert.AreEqual(CoreErrorMessages.UnsupportedAuthorityValidation, exc.Message);
-                }
+                Task.Run(
+                    async () =>
+                    {
+                        await instance.ResolveEndpointsAsync(
+                            httpManager,
+                            new TelemetryManager(),
+                            null,
+                            new RequestContext(null, new TestLogger(Guid.NewGuid(), null))).ConfigureAwait(false);
+                    }).GetAwaiter().GetResult();
+
+                Assert.AreEqual(
+                    "https://mytenant.com.b2clogin.com/6babcaad-604b-40ac-a9d7-9fd97c0b779f/oauth2/v2.0/authorize?p=my-policy",
```

Comment:
>my-policy [](start = 116, length = 9)

We should work with the function where the policy is in the url after the tenant and not added as a parameter. This is not oidc compliant.